EnhancedMediaPlayer/MediaPlayerViewController.swift

```diff
@@ -0,0 +1,73 @@
+// MediaPlayerViewController.swift
+
+import SwiftUI
+
+public class MediaPlayerViewController: UIHostingController<MediaPlayerView> {
+    private let viewModel: MediaPlayerView.ViewModel
+
+    public init(mediaURL: URL) {
+        self.viewModel = .init(mediaURL: mediaURL)
+
+        super.init(rootView: .init(viewModel: viewModel))
+
+        setupActions()
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        play()
+    }
```

Comment:
I actually think this should probably be in `viewDidAppear`, no? Or `viewWillAppear`